File: content-solutions/binary-search/eating-speed.java
class Solution {
    public int solution(int[] piles, int hours) {
        long lo = 1, hi = 0;
        for (int p : piles) hi = Math.max(hi, p);
        long answer = hi;
        while (lo <= hi) {
            long mid = (lo + hi) / 2;
            long need = 0;
            for (int p : piles) need += (p + mid - 1) / mid;
            if (need <= hours) {
                answer = mid;
                hi = mid - 1;
            } else lo = mid + 1;
        }
        return (int) answer;
    }
}

File: content-solutions/binary-search/eating-speed.js
function solution(piles, hours) {
  let lo = 1;
  let hi = Math.max(...piles);
  let answer = hi;
  while (lo <= hi) {
    const mid = Math.floor((lo + hi) / 2);
    let need = 0;
    for (const p of piles) need += Math.ceil(p / mid);
    if (need <= hours) {
      answer = mid;
      hi = mid - 1;
    } else lo = mid + 1;
  }
  return answer;
}
